src/utils/context_free_grammar_generator.cpp

```diff
@@ -30,8 +30,7 @@ context_free_grammar_generator::~context_free_grammar_generator()
 {
 }
 
-context_free_grammar_generator::context_free_grammar_generator(const std::string& source) :
-	initialized_(false)
+context_free_grammar_generator::context_free_grammar_generator(const std::string& source) throw ()
```

Comment:
This is wrong! You only use `throw()` when you want to guarantee that the function will never throw an exception, but this function is clearly throwing exceptions.